internal/gcsx/bucket_manager_test.go

```diff
@@ -0,0 +1,129 @@
+package gcsx
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage"
+	"github.com/jacobsa/gcloud/gcs"
+	"github.com/jacobsa/gcloud/gcs/gcsfake"
+	. "github.com/jacobsa/ogletest"
+	"github.com/jacobsa/timeutil"
+)
+
+func TestFlags(t *testing.T) { RunTests(t) }
+
+const TestBucketName string = "gcsfuse-default-bucket"
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type BucketManagerTest struct {
+	fakeStorageServer *fakestorage.Server
+	bucketHandle      *storage.BucketHandle
+}
+
+var _ SetUpInterface = &BucketManagerTest{}
+var _ TearDownInterface = &BucketManagerTest{}
+
+func init() { RegisterTestSuite(&BucketManagerTest{}) }
+
+func (t *BucketManagerTest) SetUp(_ *TestInfo) {
+	var err error
+	t.fakeStorageServer, err = storage.CreateFakeStorageServer([]fakestorage.Object{storage.GetTestFakeStorageObject()})
+	AssertEq(nil, err)
+
+	storageClient := &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+	t.bucketHandle, err = storageClient.BucketHandle(TestBucketName)
+	AssertEq(nil, err)
+	AssertNe(nil, t.bucketHandle)
+}
+
+func (t *BucketManagerTest) TearDown() {
+	t.fakeStorageServer.Stop()
+}
+
+func (t *BucketManagerTest) TestNewBucketManagerMethod() {
+	var nilValue *bucketManager = nil
+	storageClient := &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+
+	bucketConfig := BucketConfig{
+		BillingProject:                     "BillingProject",
+		OnlyDir:                            "OnlyDir",
+		EgressBandwidthLimitBytesPerSecond: 7,
+		OpRateLimitHz:                      11,
+		StatCacheCapacity:                  100,
+		StatCacheTTL:                       20 * time.Second,
+		EnableMonitoring:                   true,
+		debugGcs:                           true,
+		AppendThreshold:                    2,
+		TmpObjectPrefix:                    "TmpObjectPrefix",
+	}
+
+	bm := NewBucketManager(bucketConfig, nil, storageClient)
+	ExpectNe(bm, nilValue)
+}
+
+func (t *BucketManagerTest) TestSetUpGcsBucketEnableStorageClientLibraryMethod() {
+	var bm bucketManager
+	var nilBucket *storage.BucketHandle = nil
+	bm.storageHandle = &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+	bm.config.EnableStorageClientLibrary = true
+
+	Bucket, err := bm.SetUpGcsBucket(context.Background(), TestBucketName)
```

Comment:
is it possible to check if buckets are wrapped correctly i.e, bucket should be of type DebugBucket and wrapped variable should be of type RequestTrace and so on.

Reply:
No, because debugbucket and RequestTrace are unexported type of buckets from jacobsa/gcloud

Reply:
what do you mean by unexported types?

Reply:
It is not public